### DISCUSSION TITLE
Release fix for "Deprecation warning $weight: Passing a number without unit % (100) is deprecated" in a patch version?

### DISCUSSION THREAD

**Discussion by dakur** [twbs/bootstrap#37467] [Ideas] [answered]
Ref https://github.com/twbs/bootstrap/pull/37425

This is pretty annoying error and even more if you use some error emitter in your dev stack which shows error in overlay above all content on each page load. Of course, one can disable the emitting temporarily or downgrade or whatever, but..

Couldn't be the patch released in new patch version? It doesn't seem to me it has to wait until new minor..

**Comment by julien-deramond** [comment]
Hey @dakur

We are discussing it in the core team and will provide an official answer in this discussion.

Just out of curiosity, something like the following couldn't work without a patch release from Bootstrap?

```scss
@import "../node_modules/bootstrap/scss/functions";

// Override the definition of the function raising the warning
@function opaque($background, $foreground) {
  @return mix(rgba($foreground, 1), $background, opacity($foreground) * 100%);
}

@import "../node_modules/bootstrap/scss/variables";

@import "../node_modules/bootstrap/scss/maps";
@import "../node_modules/bootstrap/scss/mixins";
@import "../node_modules/bootstrap/scss/root";

// Optionally include any other parts as needed
@import "../node_modules/bootstrap/scss/utilities";
@import "../node_modules/bootstrap/scss/reboot";
@import "../node_modules/bootstrap/scss/type";
@import "../node_modules/bootstrap/scss/images";
@import "../node_modules/bootstrap/scss/containers";
@import "../node_modules/bootstrap/scss/grid";
@import "../node_modules/bootstrap/scss/helpers";

// Optionally include utilities API last to generate classes based on the Sass map in `_utilities.scss`
@import "../node_modules/bootstrap/scss/utilities/api";

// Add additional custom code here
```

**Reply by Maaaleeek** [reply]
Hi @julien-deramond 
Overriding the function woking well, but we need to modifiy the structure of projects as below:
 - Add the override function in all entry files  OR
 - Create new scss file to add the new function and import this file in all entry file OR
 - Create new scss file for bootstrap imports with override function, and this new file in all entry file 
 Anyway, but for me, I need to make one of those chose.

**Reply by dakur** [reply]
@julien-deramond That works, thanks. I think it's still worthy to make a proper patch release rather than force people to implement workarounds (and forget to remove them afterwards) but I'm OK with it, I can make a note about removing that thing on next upgrade.

**Reply by julien-deramond** [reply]
The whole core team isn't available right now so the topic is still opened. A final decision hasn't been made. Releasing a dot version (v5.2.3) is not excluded for the moment. I'll give you an update in this discussion as soon as I have the info.

**Comment by julien-deramond** [comment, marked as answer]
Good news! ~~This patch will be released this week in the v5.2.3. Stay tuned!~~
Edit: This patch is released in the [v5.2.3](https://github.com/twbs/bootstrap/releases/tag/v5.2.3).

**Reply by dakur** [reply]
It works like a charm. Thanks for reconsidering and releasing!
